node/src/components/consensus/highway_core/state.rs

```diff
@@ -372,11 +382,34 @@ impl<C: Context> State<C> {
             self.blocks.insert(hash, block);
         }
         self.units.insert(hash, unit);
-        self.update_panorama(hash, instance_id);
+
+        // Update the panorama.
```

Comment:
I preferred when it was a separate method.

Reply:
I moved in in here because if it were called in any other place it would potentially create invalid evidence: `update_panorama` required that the new unit was
* already added to the state,
* the _most recently_ added unit.

It feels safer to me to just move the code into the only place where it's valid to call it.
Alternatively I could make it a function again, but _inside_ `add_valid_unit`?

Reply:
Oh, good call then. Didn't think of that.